armory/art_experimental/defences/mp3_compression_channelized.py

```diff
@@ -0,0 +1,25 @@
+from art.defences.preprocessor import Mp3Compression
+import numpy as np
+
+
+class Mp3CompressionChannelized(Mp3Compression):
+    """
+    Add a channel axis to the input
+    """
+
+    def __init__(
+        self, sample_rate, channels_first=False, apply_fit=False, apply_predict=True,
+    ):
+        super().__init__(
+            sample_rate=sample_rate,
+            channels_first=channels_first,
+            apply_fit=apply_fit,
+            apply_predict=apply_predict,
+        )
+
+    def __call__(self, x, y=None):
+        # add a channel axis
+        x = np.expand_dims(x, axis=-1)
+        x, _ = super().__call__(x)
+        x = np.squeeze(x)
```

Comment:
@yusong-tan I believe if the batch size is 1, the squeeze operation will collapse the batch dimension which will cause errors during inference. Perhaps we could use an axis in the call to squeeze, or raise an error if the batch size is 1 for this defense?